### ISSUE TITLE
What is the state of this project?

### ISSUE THREAD

**Issue by meow464** [clayrisser/react-gtk#24]
I can see the commit rate is very low but I was able to run some examples on my machine. Will it just work™ with react-native and react-native-windows?
What are things that still need to be done?

Thanks.

**Comment by clayrisser** [comment]
I'm in the middle of rewriting it. I think I will be getting around to it towards the middle of 2021.

**Comment by clayrisser** [comment]
I'm planning on adding GTK 4 support.

**Comment by 3GOMESz** [comment]
> I'm planning on adding GTK 4 support.

Any plans for a solution to create apps with a window background blur effect in GTK4?

**Comment by 3GOMESz** [comment]
In fact, congratulations to the project, insisting on continuing the development, I am sure that the repository will gain more and more visibility.

**Comment by clayrisser** [comment]
Once I migrate to GTK 4, it should support the entire GTK4 api.

**Comment by clayrisser** [comment]
It might take me over 6 months to migrate. I'm pretty busy. I do have plans to finish this though.

**Comment by Fabsolute** [comment]
any update ?

**Comment by clayrisser** [comment]
It's kinda on pause until I can get to it. I'm really hoping I can get to it this year. I think I might start by building out a basic architecture so people interested in this can contribute. I did a lot of work and finished react-ast, a major prerequisite for this project.

https://github.com/clayrisser/react-ast

**Comment by clayrisser** [comment]
The reason I need react ast is because I need a highly composable way to generate the typescript react elements and components from the gir specs.

**Comment by clayrisser** [comment]
I would be able to get to it faster if someone was able to raise money or sponsor this project. I don't have the time to raise any money, but regardless I definitely have plans to finish this. There is a lot of interest around it.

**Comment by rockiger** [comment]
Couldn't you make it easier for others to contribute? I personally think the Gnome/GTK ecosystem is in dire need of declarative approach to app development.

**Comment by clayrisser** [comment]
@rockiger yes I definitely want to make it easier for others to contribute. I just need to take some time to set it up. I'll try and push to get it done soon.

**Comment by ghost** [comment]
Hi all!  This project is wonderful, I wish I had created a react-gtk. Guys, there is a project called mitosis. I'm thinking of integrating react-gtk into [mitosis](https://github.com/BuilderIO/mitosis ).

> "Write components once, run everywhere. Compiles to Vue, React, Solid, Angular, Svelte, and more." - mitosis

The reason for this would be to facilitate the creation of mockups, prototypes, webapps natively on linux independently of any linux distro. So... [mitosis](https://mitosis.builder.io) would be like "java", you write in [mitosis](https://mitosis.builder.io), and that can be turned into vuejs, react, gtk, etc.

**Comment by clayrisser** [comment]
I know it's been a long time since I've checked in on this project. I still have plans to rebuild this for GTK 4, which I think would be amazing since GTK 4 is now targeting mobile apps for the phosh operating system.

https://gitlab.gnome.org/World/Phosh/phosh

The main reason I have been held up on this project, is because I am finishing the development of [multiplatform.one](https://multiplatform.one), a meta framework that enables React development to target any platform. Write once, render anywhere. My plan is to bring this project into the multiplatform.one ecosystem.

https://gitlab.com/risserlabs/multiplatform.one

**Comment by NathanC** [comment]
Heya, just checking in on this. I could potentially be interested in helping-- I'm getting more into Linux/GTK, and being able to use React (and mitosis) would be amazing.

What about using React native for this? Isn't this essentially a native bridge?

**Comment by matthewruzzi** [comment]
There have been no commits in over two years. Is this project abandoned?

**Comment by clayrisser** [comment]
Not abandoned, but on hold until I get the time to rebuild it for GTK4.

**Comment by cAttte** [comment]
abandoned it is, sadly

**Comment by clayrisser** [comment]
There is hope. Someday...

**Comment by anultravioletaurora** [comment]
Hey! @clayrisser are you in need of support of this project?

I'd like to use this on our [React Native project](https://github.com/Jellify-Music/App) and can dedicate some resources in helping out! 

I'm a big fan of this project, and it's what I'd like to use for getting native desktop linux support. Our userbase for desktop would largely be Linux, so I'd see this as a huge win if we could adopt React-GTK

**Comment by nikelborm** [comment]
There's a new player in town
https://github.com/gtkx-org/gtkx

**Comment by sonnyp** [comment]
And https://gitlab.gnome.org/vixalien/peachy/
